src/common/recursive-form/boolean-input.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { Col, Row } from 'grid'
+import { autobind } from 'utils'
+
+import AbstractInput from './abstract-input'
+import { descriptionRender } from './helpers'
+
+// ===================================================================
+
+export default class BooleanInput extends AbstractInput {
+  constructor (props) {
+    super(props)
+    this.state = {
+      checked: Boolean(props.value)
+    }
+  }
+
+  get value () {
+    return this.refs.input.checked
+  }
+
+  set value (checked) {
+    checked = Boolean(checked)
+
+    this.setState({
+      checked
+    }, () => { this.refs.input.checked = Boolean(checked) })
+  }
+
+  @autobind
+  _onChange (checked) {
+    const { onChange } = this.props
+
+    this.setState({
+      checked
+    })
+
+    if (onChange) {
+      onChange(checked)
+    }
+  }
+
+  render () {
+    const { props } = this
+    const {
+      checked
+    } = this.state
+
+    return (
+      <Row>
+        <Col mediumSize={6}>
+          <div className='input-group'>
+            <div className='form-inline'>
+              <div className='checkbox' style={{ color: '#31708F' }}>
```

Comment:
It could be interesting to define this as a standalone component (maybe in `form.js`).